File: LC413-Arithmetic-Slices/code.py
class Solution:
    def numberOfArithmeticSlices(self, nums: List[int]) -> int:
        
        n = len(nums)
        if n < 3:
            return 0
        
        t_cnt, ap_cnt = 0, 0
        for i in range(0, n-2):
            if nums[i+1]-nums[i] == nums[i+2]-nums[i+1]:
                t_cnt += 1
            else:
                ap_cnt += (t_cnt)*(t_cnt+1)//2
                t_cnt = 0
        
        if t_cnt:
            ap_cnt += (t_cnt)*(t_cnt+1)//2
        
        return ap_cnt

File: Medium/LC413-Arithmetic-Slices/code.cpp
class Solution {
public:
    
    int numberOfArithmeticSlices(vector<int>& nums) {
        
        int n = nums.size();
        if(n < 3) {
            return 0;
        }

        int count_triplets = 0, count_aps = 0;
        for(int i = 0 ; i < n-2 ; i++) {
            if(nums[i+1] - nums[i] == nums[i+2] - nums[i+1]) {
                count_triplets++;
            }else {
                count_aps += (count_triplets)*(count_triplets+1)/2;
                count_triplets = 0;
            }
        }

        if(count_triplets){
            count_aps += (count_triplets)*(count_triplets+1)/2;
            count_triplets = 0;
        }
        return count_aps;
    }
};
